examples/lock-app/esp32/main/DeviceCallbacks.cpp

```diff
@@ -72,3 +76,19 @@ void AppDeviceCallbacks::OnOnOffPostAttributeChangeCallback(EndpointId endpointI
 exit:
     return;
 }
+
+void AppDeviceCallbacks::OnDoorLockPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+    VerifyOrExit(attributeId == ZCL_LOCK_STATE_ATTRIBUTE_ID, ESP_LOGI(TAG, "Unhandled Attribute ID: '0x%04x", attributeId));
+    VerifyOrExit(endpointId == 1 || endpointId == 2, ESP_LOGE(TAG, "Unexpected EndPoint ID: `0x%02x'", endpointId));
+    if (*value)
+    {
+        BoltLockMgr().InitiateAction(AppEvent::kEventType_Lock, BoltLockManager::LOCK_ACTION);
```

Comment:
Shouldn't this somehow pass through the endpoint id?  Which thing are we locking/unlocking?

Reply:
We are passing endpoint in the callback `OnDoorLockPostAttributeChangeCallback ` itself

Reply:
Yes, I know, but we have two locks here: one on endpoint 1 and one on endpoint 2.  Why are we not passing which lock the action is for to the `InitiateAction` call?

Reply:
This was redundency. Lock/Unlock action pasees through endpoints [here](https://github.com/project-chip/connectedhomeip/pull/24169/files#diff-eaa1b70782d02d5990bbfad4e85ae09441b53e9647185806e9c12840d0cfc1b0R49)

Reply:
Maybe I misunderstand what `InitiateAction` does?  Why does it not need to know which endpoint's bolt is supposed to move?

Reply:
`InitiateAction` make sure that if one action is completed then only register other action.